src/types/convert_util.rs

```diff
@@ -31,29 +31,77 @@ impl From<compositor_render::Resolution> for Resolution {
+
+        let interpolation_kind = match easing_function {
+            EasingFunction::Linear => scene::InterpolationKind::Linear,
+            EasingFunction::Ease => scene::InterpolationKind::Ease,
+            EasingFunction::EaseIn => scene::InterpolationKind::EaseIn,
+            EasingFunction::EaseOut => scene::InterpolationKind::EaseOut,
+            EasingFunction::EaseInOut => scene::InterpolationKind::EaseInOut,
+            EasingFunction::EaseInQuint => scene::InterpolationKind::EaseInQuint,
+            EasingFunction::EaseOutQuint => scene::InterpolationKind::EaseOutQuint,
+            EasingFunction::EaseInOutQuint => scene::InterpolationKind::EaseInOutQuint,
+            EasingFunction::EaseInExpo => scene::InterpolationKind::EaseInExpo,
+            EasingFunction::EaseOutExpo => scene::InterpolationKind::EaseOutExpo,
+            EasingFunction::EaseInOutExpo => scene::InterpolationKind::EaseInOutExpo,
+            EasingFunction::Bounce => scene::InterpolationKind::Bounce,
+            EasingFunction::CubicBezier { points } => {
+                if points[0] < 0.0 || points[0] > 1.0 {
+                    return Err(TypeError::new(
+                        "Control point x1 has to be in the range [0, 1].",
+                    ));
+                }
+                if points[2] < 0.0 || points[2] > 1.0 {
+                    return Err(TypeError::new(
+                        "Control point x2 has to be in the range [0, 1].",
+                    ));
+                }
+
+                scene::InterpolationKind::CubicBezier {
+                    x1: points[0],
+                    y1: points[1],
+                    x2: points[2],
+                    y2: points[3],
+                }
+            }
         };
 
-        Self {
+        Ok(Self {
             duration: Duration::from_secs_f64(transition.duration_ms / 1000.0),
             interpolation_kind,
+        })
+    }
+}
+
+impl FromStr for EasingFunction {
+    type Err = TypeError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "linear" => Ok(Self::Linear),
+            "ease" => Ok(Self::Ease),
+            "ease_in" => Ok(Self::EaseIn),
+            "ease_out" => Ok(Self::EaseOut),
+            "ease_in_out" => Ok(Self::EaseInOut),
+            "ease_in_quint" => Ok(Self::EaseInQuint),
+            "ease_out_quint" => Ok(Self::EaseOutQuint),
+            "ease_in_out_quint" => Ok(Self::EaseInOutQuint),
+            "ease_in_expo" => Ok(Self::EaseInExpo),
+            "ease_out_expo" => Ok(Self::EaseOutExpo),
+            "ease_in_out_expo" => Ok(Self::EaseInOutExpo),
+            "bounce" => Ok(Self::Bounce),
+            "cubic_bezier" => Err(TypeError::new(
+                "Cubic bezier easing function needs control points specified. Please define it as an object with the control points provided."
+            )),
+            _ => Err(TypeError::new("Unknown easing function.")),
```

Comment:
I'm not sure how to implement it better

Reply:
Don't check it on serialization. Use a separate type for API and one for implementation. Put this logic in From<APItype> for BackendImplType and return type error on scene update. With 2 types this check is also forced by the type system and you don't have to write this weird deserialization. We do it like this in other places, so I would keep that approach as a project convention.

Reply:
The problem is that the enum value can be either represented as a string (for example: `"linear"`) or as an object:
```json
{
    "function_name": "cubic_bezier",
    "points": [0.0, 0.0, 1.0, 1.0]
}
```

I could simplify it by using only the object notation.

Reply:
Not if you use:
```rust
#[derive(Serialize, Deserialize)]
#[serde(untagged)]
enum StringOrObject {
    String(String)
    Object(Object)
}
```

Reply:
So it would look like:
```rust
#[derive(Serialize, Deserialize)]
#[serde(untagged)]
enum ApiBezierParams {
    BuiltIn(BuiltInInterpolations)
    Custom(CustomBezier)
}

#[derive(Serialize, Deserialize)]
enum BuiltIn {
    Linear,
    Ease,
    EaseIn,
    EaseOut,
    EaseInOut,
    EaseInQuint,
    EaseOutQuint,
    EaseInOutQuint,
    EaseInExpo,
    EaseOutExpo,
    EaseInOutExpo,
    Bounce,
}

#[derive(Serialize, Deserialize)]
struct CustomBezier {
    points: [f64; 4]
}
```